lib/SearchAndSort/SearchAndSort.js

```diff
@@ -239,10 +239,20 @@ class SearchAndSort extends React.Component {
   }
 
   componentDidMount() {
-    const query = this.queryParam('query');
-    this.setState({
-      locallyChangedSearchTerm: query
-    });
+    const {
+      location: {
+        search,
+      },
+      browseOnly,
+    } = this.props;
+    const resQuery = this.queryParam('query');
+    const { query } = queryString.parse(search);
+
+    // update locallyChangedSearchTerm only
+    // when query from resourceQuery matches url query
+    const term = (browseOnly && resQuery === query) ? query : '';
+
+    this.setState({ locallyChangedSearchTerm: term });
```

Comment:
The code looks good @mkuklis. I am just wondering about moving the logic itself to the method and just make a call in the component did mount. What do you think about that?